### PR TITLE
Remove dependency on ICompletionSession and ISignatureHelpSession

### LINKED ISSUE #20525: AsyncCompletionService shouldn't depend on VS Editor's ICompletionSession
http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/Intellisense/Completion/AsyncCompletionService.cs,28

```
private readonly IIntelliSensePresenter<ICompletionPresenterSession, ICompletionSession> _completionPresenter;
```

We have an IIntelliSensePresenter that depends on ICompletionSession which is an Editor concept. We need to find a way to remove this since it's only used by the Quick Info implementation. This prevents the Visual Studio for Mac team from implementing our own IIntelliSensePresenter.

cc @toddgrun

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #20525

### CODE REVIEW

File: src/EditorFeatures/Core/CommandHandlers/QuickInfoCommandHandlerAndSourceProvider.cs [hunk 1]
```diff
@@ -35,17 +35,17 @@ internal partial class QuickInfoCommandHandlerAndSourceProvider :
         public QuickInfoCommandHandlerAndSourceProvider(
             [ImportMany] IEnumerable<Lazy<IQuickInfoProvider, OrderableLanguageMetadata>> providers,
             [ImportMany] IEnumerable<Lazy<IAsynchronousOperationListener, FeatureMetadata>> asyncListeners,
-            [ImportMany] IEnumerable<Lazy<IIntelliSensePresenter<IQuickInfoPresenterSession, IQuickInfoSession>, OrderableMetadata>> presenters)
-            : this(ExtensionOrderer.Order(presenters).Select(lazy => lazy.Value).FirstOrDefault(),
+            [ImportMany] IEnumerable<Lazy<IIntelliSensePresenter<IQuickInfoPresenterSession>, OrderableMetadata>> presenters)
+            : this(ExtensionOrderer.Order(presenters).Select(lazy => lazy.Value).OfType<IIntelliSensePresenter<IQuickInfoPresenterSession, IQuickInfoSession>>().FirstOrDefault(),
```
Comment (Therzok) [inline src/EditorFeatures/Core/CommandHandlers/QuickInfoCommandHandlerAndSourceProvider.cs]: Isn't this just masking the API dependency? You wouldn't be able to define a quick info presenter now.

Comment (sharwell) [inline src/EditorFeatures/Core/CommandHandlers/QuickInfoCommandHandlerAndSourceProvider.cs]: I removed the dependencies on `ICompletionSession` and `ISignatureHelpSession`. I did not remove the dependency on `IQuickInfoSession`.

Comment (Therzok) [inline src/EditorFeatures/Core/CommandHandlers/QuickInfoCommandHandlerAndSourceProvider.cs]: :+1:

**Comment by sharwell** [comment]
@dotnet-bot retest this please

**Comment by cston** [comment]
@dotnet-bot test windows_debug_vs-integration_prtest please

**Comment by sharwell** [comment]
@dotnet-bot retest windows_release_vs-integration_prtest please

**Pull request closed without merging**